lib/iotagent-json.js

```diff
@@ -137,7 +137,7 @@ function start(newConfig, callback) {
                 iotAgentLib.addQueryMiddleware(iotAgentLib.dataPlugins.compressTimestamp.query);
             }
 
-            if (config.getConfig().mqtt.thinkingThingsPlugin) {
+            if (config.getConfig().mqtt && config.getConfig().mqtt.thinkingThingsPlugin) {
```

Comment:
I understand this is not related to the topic of the PR, right?

(It's fine adding it, just asking confirmation of scope)

Reply:
Right, I discover this bug making some tests. Due to it is so tricky to resolve, I decided to include in this PR.